### DISCUSSION TITLE
Best practice for postgres functions

### DISCUSSION THREAD

**Discussion by arshamg** [supabase/supabase#11234] [Questions]
What is the best practice for Postgres functions and Supabase? Do people generally store them under version control? It would be great if there was a way to deploy them from a Github Action or similar.

**Comment by davepar** [comment]
I've been storing functions and other stuff (policies, enum definitions, triggers, views, etc.) in files in git next to my client code. So far I've just been copy/pasting into the SQL editor in the Supabase UI, but that's not a great workflow.

With Postgres at the center of the Supabase platform and playing such a major role, this seems to be a real shortcoming, i.e. how to manage and migrate changes to the database. They seem to support only one workflow: develop on Supabase local (which doesn't support all of the features) and then migrate changes to a hosted Supabase project via the CLI. There's another discussion related to this: https://github.com/supabase/supabase/discussions/11212
